### PR TITLE
feat: adds class, custom exception, and test file 

### PR DESCRIPTION
**Pull request by RichHawkins**

### CODE REVIEW

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package com.codedifferently.lesson7.richhawkins;
+
+import java.time.Year;
+
+public class House {
+  //     The class must have at least 5 member variables of at least 3 different types.
+  String streetName;
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java:7]: your member variables should be `private`.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  int streetNum;
+  int yearBuilt;
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java:9]: You're not using the `yearBuilt`, `hasGarage`, and `numberOfBedrooms` members. You should create and test property accessors for them or remove.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  public House(String streetName, int streetNum) {
+    this.streetName = streetName;
+    this.streetNum = streetNum;
+  }
+
+  // The class must have at least 3 member functions.
+
+  // One of your functions must make use of a conditional expression.
+  // You must use at least one custom exception.
+  public boolean checkValidYear(int yearBuilt) {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java:40]: I'd expect this method to be `private` and to be used in a setter property called `setYearBuilt` that allows you to set a proper year or throw an exception if the year is not valid. Please revise.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java (hunk 1 above)
Comment (RichHawkins) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java:7]: member variables set to private. Created getters and setter so the variables are still accessible.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java (hunk 2 above)
Comment (RichHawkins) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java:9]: All member variables now used with property accessors (learned what that meant) and tests are included for them.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java (hunk 3 above)
Comment (RichHawkins) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java]: This really helped make a few things click for me. Should be revised as requested.

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/richhawkins/HouseTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  @Test
+  public void testSetYearBuilt() {
+    House house = new House("Pine", 234);
+
+    house.setYearBuilt(2020);
+    assertEquals(2020, house.getYearBuilt());
+
+    try {
+      house.setYearBuilt(2030);
+      fail("Exception not thrown");
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/richhawkins/HouseTest.java:20]: Use `assertThatThrownBy` (preferred) or `assertThrows` to test that an exception has been thrown.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  public String printFullAddress() {
+    String address = ("Your address is " + streetNum + " " + streetName);
+    return address;
+  }
+
+  // One of your functions must make use of a loop.
+  public String allHouseType() {
+    StringBuilder types = new StringBuilder();
+    for (HouseType type : HouseType.values()) {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java:96]: Needed to include a collection type. Enum values almost count.

**Review by anthonydmays — APPROVED** [review]
LGTM

**Pull request merged**